src/HasStates.php

```diff
@@ -57,11 +66,16 @@ public static function bootHasStates(): void
                     ? new $stateConfig->defaultStateClass($model)
                     : null;
 
+                /** @var \Spatie\ModelStates\State $state */
+                $state = class_exists($stateClass)
+                    ? new $stateClass($model)
+                    : $defaultState;
+
+                $state->setField($stateConfig->field);
```

Comment:
Sometimes, your `$defaultState` variable [can be null](https://github.com/spatie/laravel-model-states/pull/31/commits/c3002659d09166c7df9b3634641c44e2c4345b75#diff-32ee99deed04fb3f9b53df25767954eaR67) if there is no default state class in state configuration object

I suggest something like this:

```php
/** @var null|\Spatie\ModelStates\State $state */
$state = $defaultState;

if(class_exists($stateClass)){
    $state = new $stateClass($model);
    $state->setField($stateConfig->field);
}
```

Reply:
Good catch, thanks!

Reply:
You didn't removed code duplication on [line 78](https://github.com/spatie/laravel-model-states/pull/31/commits/661d4c5e0dbc8ac7372f7b1058e80c96582ef6a6#diff-32ee99deed04fb3f9b53df25767954eaR78)

Reply:
I fixed it in the v2 branch :)